### PR TITLE
Update docs with nomenclature changes

### PR DESCRIPTION
**Pull request by devashish-patel**

### CODE REVIEW

File: website/content/docs/hcp/index.mdx [hunk 1]
```diff
@@ -1,72 +1,120 @@
...
+The HCP Packer registry bridges the gap between artifact factories and artifact deployments, allowing development and
+security teams to work together to create, manage, and consume artifacts in a centralized way.
 
-The HCP Packer registry stores metadata about your images, including when they were created, where the image exists in the cloud, and what (if any) git commit is associated with your image build. You can use the registry to track information about the golden images your Packer builds produce, clearly designate which images are appropriate for test and production environments, and query for the right golden images to use in both Packer and Terraform configurations.
+The HCP Packer Registry stores metadata about your artifact, including when they were created, where the artifact
+exists on the external platform, and what (if any) git commit is associated with your build. You can use the registry
+to track information about the artifact your Packer builds produce, clearly designate which artifact are appropriate
+for test and production environments, and query for the right golden artifact to use in both Packer and Terraform
```
Comment (nywilken) [inline website/content/docs/hcp/index.mdx:21]: Just artifact. Golden artifact is not a thing ppl know. 
```suggestion
for test and production environments, and query for the right artifact to use in both Packer and Terraform
```

File: website/content/docs/datasources/hcp/hcp-packer-version.mdx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+data "hcp-packer-version" "hardened-source" {
+  bucket_name  = "hardened-ubuntu-16-04"
+  channel_name = "dev"
+}
+
+# Retrieves information about the HCP Packer Artifact; an artifact can be thought
+# of as all the metadata (including the artifact names) created by a single
+# "source" builder; this can include multiple artifacts so we provide a
+# region to disambiguate.
+data "hcp-packer-artifact" "foo" {
```
Comment (nywilken) [inline website/content/docs/datasources/hcp/hcp-packer-version.mdx:65]:
```suggestion
data "hcp-packer-artifact" "example" {
```

File: website/content/docs/datasources/hcp/hcp-packer-version.mdx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  version_fingerprint = data.hcp_packer_version.hardened-source.fingerprint
+  platform            = "aws"
+  region              = "us-east-1"
+}
+
+# This source uses the output from a previous Packer build. By using the
+# HCP Packer Registry in this way, you can easily create build pipelines where
+# a single base artifact can be customized in multiple secondary layers.
+source "amazon-ebs" "packer-secondary" {
+  source_ami       = data.hcp-packer-artifact.foo.id
```
Comment (nywilken) [inline website/content/docs/datasources/hcp/hcp-packer-version.mdx:76]: Lets try not to use foo in the example. Use either example or a name that people would use in the template to describe the infrastructure as code. 
```suggestion
  source_ami       = data.hcp-packer-artifact.example.id
```

File: website/content/docs/datasources/hcp/hcp-packer-artifact.mdx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+data "hcp-packer-version" "hardened-source" {
+  bucket_name  = "hardened-ubuntu-16-04"
+  channel_name = "dev"
+}
+
+# Retrieves information about the HCP Packer Artifact; an artifact can be thought
+# of as all the metadata (including the artifact names) created by a single
+# "source" builder; this can include multiple artifacts so we provide a
+# region to disambiguate.
+data "hcp-packer-artifact" "foo" {
```
Comment (nywilken) [inline website/content/docs/datasources/hcp/hcp-packer-artifact.mdx:68]:
```suggestion
data "hcp-packer-artifact" "example" {
```

File: website/content/docs/datasources/hcp/hcp-packer-artifact.mdx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  version_fingerprint = data.hcp_packer_version.hardened-source.fingerprint
+  platform            = "aws"
+  region              = "us-east-1"
+}
+
+# This source uses the output from a previous Packer build. By using the
+# HCP Packer Registry in this way, you can easily create build pipelines where
+# a single base artifact can be customized in multiple secondary layers.
+source "amazon-ebs" "packer-secondary" {
+  source_ami       = data.hcp-packer-artifact.foo.id
```
Comment (nywilken) [inline website/content/docs/datasources/hcp/hcp-packer-artifact.mdx:79]:
```suggestion
  source_ami       = data.hcp-packer-artifact.example.id
```

File: website/data/docs-nav-data.json [hunk 6]
```diff
@@ -730,11 +730,19 @@
...
-            "title": "Iteration",
+            "title": "Version",
+            "path": "datasources/hcp/hcp-packer-version"
+          },
+          {
+            "title": "Artifact",
+            "path": "datasources/hcp/hcp-packer-artifact"
+          },
+          {
+            "title": "Iteration (Deprecated)",
```
Comment (nywilken) [inline website/data/docs-nav-data.json:741]: I'm not sure how I feel about having Deprecated in the nav bar but lets see what it looks like.

**Review by nywilken — COMMENTED** [review]
The contents of the docs and examples looks good. I left a few suggested changes.

File: website/content/docs/datasources/hcp/hcp-packer-artifact.mdx [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+the build. Building new artifacts from a revoked artifact is not compliant.
+Versions that are scheduled to be revoked will still be considered valid until the revocation date.
+
+## Basic Example
+
+Below is a fully functioning example. It stores information about an image,
+which can then be parsed and accessed as a variable.
+
+```hcl
+data "hcp_packer_artifact" "example" {
```
Comment (nywilken) [inline website/content/docs/datasources/hcp/hcp-packer-artifact.mdx:41]:
```suggestion
data "hcp-packer-artifact" "example" {
```

**Review by devashish-patel — COMMENTED** [review]
Add PR suggestions and some other changes

**Comment by nywilken** [comment]
This is ready to be rebased, squashed and merged into the hcp_packer_nomenclature_updates branch.

**Review by nywilken — APPROVED** [review]

**Pull request merged**